src/coreclr/jit/gentree.h

```diff
@@ -559,6 +559,8 @@ enum GenTreeFlags : unsigned int
 #if defined(TARGET_XARCH) && defined(FEATURE_HW_INTRINSICS)
     GTF_HW_EM_OP                  = 0x10000000, // GT_HWINTRINSIC -- node is used as an operand to an embedded mask
 #endif // TARGET_XARCH && FEATURE_HW_INTRINSICS
+
+    GTF_CONVERSION_SATURATED    = 0x20000000, // GT_CAST -- conversion operation has saturation behavior
```

Comment:
Do we need this new flag? Aren't all of the floating-point GT_CAST nodes going to simply be saturated all the time now?

Reply:
I needed some flag checking because a cast node is morphed twice. Once when the actual cast reaches morph (There all the nodes needed for saturation are added and it is sent to morph again). Second when the cast is morphed after adding nodes needed for saturation. Hence, A flag was added to track if this addition of extra nodes is already done. 

Below code shows the handling where this flag is used.
https://github.com/dotnet/runtime/pull/97529/files#diff-5b83397bbbdd17bb9457998b520fdaaa474d165390985b66f32371561b6d0bacR337-R339

Reply:
Do we need to remorph the node the second time? Why not just mark it as morphed and not recurse (I'd expect no transforms to happen the second time).

Reply:
I have not investigated it in this direction. I will investigate and update accordingly. The reason being, I am adding new nodes and I did not investigate how they are morphed. I wanted to let JIT take care of that. I will update once done.

Reply:
I think this flag is still unnecessary.